### PR TITLE
[Embed API] Ruby Client Library and Tests

### PR DESCRIPTION
**Pull request by charleslai**
Ruby client library to make calls to the new Embed API. Added function calls and tests.

### CODE REVIEW

**Comment by charleslai** [comment]
nondeterministic failures are the best kind of failures /s

File: lib/dogapi/facade.rb [hunk 1]
```diff
@@ -296,6 +297,29 @@ def graph_snapshot(metric_query, start_ts, end_ts, event_query = nil)
...
+    #
+    def get_all_embeds()
+      @embed_svc.get_all_embeds()
+    end
+
+    def get_embed(embed_id, size="medium", legend="no", template_vars= {})
+      @embed_svc.get_embed(embed_id, size, legend, template_vars)
+    end
+
+    def create_embed(graph_json, timeframe="1_hour", size="medium", legend="no", title="Embed created through API")
```
Comment (yannmh) [inline lib/dogapi/facade.rb]: Is the `title` parameter mandatory ? 

From [documentation](http://docs.datadoghq.com/api/#embeds) page, it looks like it's automatically set when no value is submitted, i.e.:
```
curl -POST \
    -d 'graph_json={"requests":[{"q":"avg:system.load.1{*}"}],"viz":"timeseries","events":[]}' \
    -d "timeframe=1_hour" \
    -d "size=medium" \
    -d "legend=yes" \
    "https://app.datadoghq.com/api/v1/graph/embed?api_key=${api_key}&application_key=${app_key}"
```
leads to 
```
{   
  "embed_id": "5f585b01c81b12ecdf5f40df0382738d0919170639985d3df5e2fc4232865b0c", 
  "template_variables": [], 
  "html": "<iframe src=\"https://app.datadoghq.com/graph/embed?token=5f585b01c81b12ecdf5f40df0382738d0919170639985d3df5e2fc4232865b0c&height=300&width=600&legend=true\" width=\"600\" height=\"300\" frameBorder=\"0\"></iframe>", 
  "graph_title": "Embed created through API", 
  "revoked": false, 
  "dash_url": null, 
  "shared_by": 3658, 
  "dash_name": null
```
(note the `graph_title` field value)

Comment (charleslai) [inline lib/dogapi/facade.rb]: It isn't required. Does I make title required in the Ruby code? I'm not super familiar with Ruby so I assumed the default parameters work the same way as they do in Python

Comment (charleslai) [inline lib/dogapi/facade.rb]: Well titles are required in the sense that the embed has to have a title at least 1 character in length and the default title is `"Embed created through API"`

Comment (charleslai) [inline lib/dogapi/facade.rb]: I guess the default params can be `nil`

**Comment by yannmh** [comment]
It looks great overall. 

**However**, I think we should make all optional parameters part of a *kwarg*, like we do for most of our API features (unfortunately not all :-1: ).
i.e having
```
dog.create_embed(graph_json, :timeframe => timeframe, :size => size, :legend => legend ...)
```
rather than
```
dog.create_embed(graph_json, timeframe, size, legend, title)
```

**Comment by charleslai** [comment]
Change parameters to use a dictionary/hash kwargs

**Comment by yannmh** [comment]
Thanks for the update @charleslai. 

I added two nitpicks. Could you also update the [documentation](https://github.com/DataDog/documentation) to reflect those changes please ?

File: tests/test_embed.rb [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      :timeframe => timeframe,
+      :size => size,
+      :legend => legend,
+      :title => title
+    }
+    status, result = dog.create_embed(graph_json, description)
+    # Sanity check results
+    assert result["graph_title"] == title
+    assert_equal status, "200", "invalid HTTP response => #{status}"
+  end
```
Comment (yannmh) [inline tests/test_embed.rb]: Can you check at least one of the parameters in the `description` ?

File: tests/test_embed.rb [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      :size => size,
+      :legend => legend,
+      :var => "val"
+    }
+    status, embed_with_var = dog.get_embed(embed_id, get_embed_description)
+    assert_equal status, "200", "invalid HTTP response => #{status}"
+    # Verify that the get requests are good
+    assert embed_without_var["html"] == create_result["html"]
+    assert embed_with_var["html"] != embed_without_var["html"]
+  end
```
Comment (yannmh) [inline tests/test_embed.rb]: Can you check that the values returned match the `description` ?

File: tests/test_embed.rb [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    description = {
+      :timeframe => timeframe,
+      :size => size,
+      :legend => legend,
+      :title => title
+    }
+    status, result = dog.create_embed(graph_json, description)
+    # Sanity check results
+    assert result["graph_title"] == title
+    assert_equal status, "200", "invalid HTTP response => #{status}"
```
Comment (yannmh) [inline tests/test_embed.rb]: Should we check that the *embed* created match the description ?

Comment (charleslai) [inline tests/test_embed.rb]: you can but you would have to parse the html though which is kind of annoying. but I can add it if you think it's worth adding

File: tests/test_embed.rb (hunk 3 above)
Comment (charleslai) [inline tests/test_embed.rb]: I sort of do a check by checking if the html after using get is different from the html that was originally created but I guess I could try parsing the html string

File: tests/test_embed.rb (hunk 4 above)
Comment (charleslai) [inline tests/test_embed.rb]: added some substring tests

**Comment by yannmh** [comment]
Great work. Thanks for addressing all the comments.

I am waiting for the tests to complete and merging.

**Pull request merged**